flink-end-to-end-tests/flink-tpcds-test/src/main/java/org/apache/flink/table/tpcds/TpcdsTestProgram.java

```diff
@@ -0,0 +1,187 @@
+							.toArray(new TypeInformation[0])
+					));
+			tableEnvironment.insertInto(resultTable, sinkTableName);
+			tableEnvironment.execute(queryName);
+			System.out.println("[INFO]Run TPC-DS query " + queryId + " success.");
+		}
+	}
+
+	/**
+	 * Prepare TableEnvironment for query.
+	 *
+	 * @param sourceTablePath
+	 * @return
+	 */
+	private static TableEnvironment prepareTableEnv(String sourceTablePath, Boolean useTableStats) throws Exception {
+		//init Table Env
+		EnvironmentSettings environmentSettings = EnvironmentSettings
+			.newInstance()
+			.useBlinkPlanner()
+			.inBatchMode()
+			.build();
+		TableEnvironment tEnv = TableEnvironment.create(environmentSettings);
+
+		//config Optimizer parameters
+		tEnv.getConfig().getConfiguration()
+			.setString(ExecutionConfigOptions.TABLE_EXEC_SHUFFLE_MODE, ShuffleMode.BATCH.toString());
+		tEnv.getConfig().getConfiguration()
+			.setString(ExecutionConfigOptions.TABLE_EXEC_RESOURCE_HASH_AGG_MEMORY, "32 mb");
+		tEnv.getConfig().getConfiguration()
+			.setString(ExecutionConfigOptions.TABLE_EXEC_RESOURCE_HASH_JOIN_MEMORY, "32 mb");
+		tEnv.getConfig().getConfiguration()
+			.setString(ExecutionConfigOptions.TABLE_EXEC_RESOURCE_SORT_MEMORY, "32 mb");
+		tEnv.getConfig().getConfiguration()
+			.setInteger(ExecutionConfigOptions.TABLE_EXEC_RESOURCE_DEFAULT_PARALLELISM, 4);
+		tEnv.getConfig().getConfiguration()
+			.setLong(OptimizerConfigOptions.TABLE_OPTIMIZER_BROADCAST_JOIN_THRESHOLD, 10 * 1024 * 1024);
+		tEnv.getConfig().getConfiguration()
+			.setBoolean(OptimizerConfigOptions.TABLE_OPTIMIZER_JOIN_REORDER_ENABLED, true);
+
+		//register TPC-DS tables
+		TCPDS_TABLES.forEach(table -> {
+			TpcdsSchema schema = TpcdsSchemaProvider.getTableSchema(table);
+			CsvTableSource.Builder builder = CsvTableSource.builder();
+			builder.path(sourceTablePath + FILE_SEPARATOR + table + DATA_SUFFIX);
+			for (int i = 0; i < schema.getFieldNames().size(); i++) {
+				builder.field(
+					schema.getFieldNames().get(i),
+					TypeConversions.fromDataTypeToLegacyInfo(schema.getFieldTypes().get(i)));
+			}
+			builder.fieldDelimiter(COL_DELIMITER);
+			builder.lineDelimiter("\n");
+			CsvTableSource tableSource = builder.build();
+			ConnectorCatalogTable catalogTable = ConnectorCatalogTable.source(tableSource, true);
+			try {
+				tEnv.getCatalog(tEnv.getCurrentCatalog()).get()
+					.createTable(new ObjectPath(tEnv.getCurrentDatabase(), table), catalogTable, false);
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		});
+		// register statistics info
+		if (useTableStats) {
+			TpcdsStatsProvider.registerTpcdsStats(tEnv);
+		}
+		return tEnv;
+	}
+
+	private static String loadFile2String(String filePath) {
+		StringBuilder stringBuilder = new StringBuilder();
+		try {
+			Stream<String> stream = Files.lines(Paths.get(filePath), StandardCharsets.UTF_8);
+			stream.forEach(s -> stringBuilder.append(s).append("\n"));
+		} catch (IOException e) {
+			e.printStackTrace();
```

Comment:
I think you want the test to fail when an exception comes, so let the exception propagate and don't swallow it.

Reply:
I missed here,  it  should be thrown